### PR TITLE
package xorg-libxinerama

### LINKED ISSUE #7197: package libximerna
I would like to package libximerna that I need to compile for glfw.

I wrote a package for libximerna that compiles on linux
https://github.com/ramonaoptics/libximerna-feedstock

A few questions:
1. Does conda forge want this package?
2. Do I need to make it work on Windows and OSX?
3. What should the package be named? `xorg-libximerna`?

### PR DESCRIPTION
**Pull request by hmaarrfk**
I don't know if we want a Windows/OSX package for this. Unless an other project requires this library, it seems tricky to compile it for them and to test that it actually works. I've tested this with GLFW and it seems to work on linux :/

 Fixes #7197

CC: @ngoldbaum  since they encouraged me to submit the PR.

### CODE REVIEW

**Comment by jakirkham** [comment]
@pkgw, would you be able to review this one?

File: recipes/xorg-libximerna/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{% set version = "1.1.4" %}
+
+package:
+  name: xorg-libximerna
```
Comment (pkgw) [inline recipes/xorg-libximerna/meta.yaml]: This package name is mistaken! It is "Xinerama" (like Cinerama), not "Ximerna". This appears in several places.

File: recipes/xorg-libximerna/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  version: {{ version }}
+
+source:
+  url: https://gitlab.freedesktop.org/xorg/lib/libxinerama/-/archive/libXinerama-{{ version }}/libxinerama-libXinerama-{{ version }}.tar.gz
+  sha256: d9fb7f224281f5467d36f54b01a8f49e866a994410d5bce3b019f2da8bf8e643
+
+build:
+  number: 1
```
Comment (pkgw) [inline recipes/xorg-libximerna/meta.yaml]: The build number should start at 0.

File: recipes/xorg-libximerna/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  skip: True   # [not linux]
+  run_exports:
+    - {{ pin_subpackage('xorg-libximerna', max_pin='x.x') }}
```
Comment (pkgw) [inline recipes/xorg-libximerna/meta.yaml]: Another "ximerna".

File: recipes/xorg-libximerna/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  commands:
+    - test -f $PREFIX/lib/libXinerama.so
+    - conda inspect linkages -p $PREFIX $PKG_NAME
+
+about:
+  home: https://gitlab.freedesktop.org/xorg/lib/libxinerama
+  license: MIT
+  license_family: MIT
+  license_file: COPYING
+  summary: 'API for Xinerama extension to X11 Protocol.'
```
Comment (pkgw) [inline recipes/xorg-libximerna/meta.yaml]: I see that this summary comes from the library README, but I don't think it's great. I'd put something more like "Client library for the Xinerama extension to the X11 protocol."

File: recipes/xorg-libximerna/build.sh [hunk 5]
```diff
@@ -0,0 +1,7 @@
+set -ex
+
+export ACLOCAL="aclocal -I ${PREFIX}/share/aclocal"
+./autogen.sh --prefix ${PREFIX} --disable-staticc
```
Comment (pkgw) [inline recipes/xorg-libximerna/build.sh]: `staticc` is probably a typo for `static`?

File: recipes/xorg-libximerna/build.sh [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+make install
+
+set +ex
```
Comment (pkgw) [inline recipes/xorg-libximerna/build.sh]: This command doesn't do anything since the shell is just going to exit anyway.

Comment (hmaarrfk) [reply recipes/xorg-libximerna/build.sh]: I guess previously with travis that wasn't so true. But maybe I am wrong. I just removed this line.

File: recipes/xorg-libximerna/meta.yaml (hunk 1 above)
Comment (hmaarrfk) [reply recipes/xorg-libximerna/meta.yaml]: yeah. I clearly can't read.

File: recipes/xorg-libximerna/meta.yaml (hunk 2 above)
Comment (hmaarrfk) [reply recipes/xorg-libximerna/meta.yaml]: Sorry, this was adapted from my personal scripts.

File: recipes/xorg-libximerna/meta.yaml (hunk 4 above)
Comment (hmaarrfk) [reply recipes/xorg-libximerna/meta.yaml]: Thanks. I really don't know what this library does, but it helped me compile my other library.

**Comment by hmaarrfk** [comment]
Hi @pkgw, Thanks for the review. I suspect that libxinerama is pretty old an obscure at this point. 
probably why it hasn't been packaged until now. 
GLFW, which uses it, doesn't include it in non-linux builds so :/

I'm happy to try and adapt an xorg-family recipe if you can point me to one. 
Would this be a good starting point?
https://github.com/conda-forge/xorg-xproto-feedstock/blob/master/recipe/build.sh

cc @EricThomson

**Comment by pkgw** [comment]
@hmaarrfk A better template would be a recipe that packages a different shared library, such as [xorg-libxi](https://github.com/conda-forge/xorg-libxi-feedstock/tree/master/recipe).

It comes with a `bld.bat` as well as the other files. The files look complex, but most of what they contain is boilerplate — although I suspect it will be somewhat challenging for a newcomer to figure out which parts are which. If I'm reading this right, the only things that need customization are the build and host requirements in `meta.yaml`, the `set lib_idents` line in `meta.yaml (line 49 for libXi), the `configure_args` section of `build.sh` (also line 49), and the `rm -rf` line there (line 62). The `bld.bat` file just delegates to `build.sh` so it has no customization.

**Comment by hmaarrfk** [comment]
@pkgw so much boilerplate. Copy paste mostly worked to my surprise!

**Comment by jakirkham** [comment]
@pkgw, would you have a chance to give this another review?

**Comment by jakirkham** [comment]
Thanks @hmaarrfk and @pkgw! 😀

**Comment by hmaarrfk** [comment]
Thanks you two!

**Pull request merged**